src/components/TransactionHistoryRows/components/AssetsTransfers.tsx

```diff
@@ -0,0 +1,55 @@
+import { Box, Circle, Stack, Text as CText, useColorModeValue } from '@chakra-ui/react'
+import { Amount } from 'components/Amount/Amount'
+import type { Transfer } from 'hooks/useTxDetails/useTxDetails'
+import { bn, bnOrZero } from 'lib/bignumber/bignumber'
+import { fromBaseUnit } from 'lib/math'
+
+import { FALLBACK_PRECISION } from '../constants'
+
+type AssetsTransfersProps = {
+  compactMode?: boolean
+  transfers: Transfer[]
+  index: number
+}
+
+export const AssetsTransfers: React.FC<AssetsTransfersProps> = ({
+  index,
+  compactMode,
+  transfers,
+}) => {
+  const circleBgColor = '#333d59' // TODO: can we reuse the Button.theme here?
+  const circleColor = useColorModeValue('blue.100', 'blue.200')
+  const aggregatedFiatValue = transfers
+    .reduce((acc, transfer) => {
+      if (!transfer) return acc
```

Comment:
We should never end up here in theory, but always wear something for safety

Reply:
If `transfer` really is a `Transfer`, an object, this should indeed never happen.

However, if we suspect it _could_ happen, and this line tells me we do, do we need to update the type of `transfers` from `Transfer[]` to `(Transfer | undefined)[]` and then filter out any `undefined` before reducing?

I'm wary of run-time guards that don't match types. If we need to guard against it we should make that explicit via the type system.

Reply:
I didn't see any occurrences of undefined `transfer`, this really just was paranoia - agreed runtime shouldn't accommodate for this and if that was even an issue, then we could just filter out truthy `transfer`s before reducing them !
Removed in 1a451ad17e15651fe9dd34d4358b026f63478133